conn_other.go

```diff
@@ -5,8 +5,13 @@ package dbus
 import (
 	"bytes"
 	"errors"
+	"fmt"
+	"io/ioutil"
 	"os"
 	"os/exec"
+	"os/user"
+	"path"
+	"strings"
 )
 
 func getSessionBusPlatformAddress() (string, error) {
```

Comment:
Should we be using 'dbus-launch --autolaunch' here instead of rolling our own logic for this?

Reply:
Thanks @jsouthworth - your comment has showed up on a funny line, can I check that you're suggesting that `dbus-launch --autolaunch` may make the `tryDiscoverDbusSessionBusAddress` function unnecessary?

If so, good steer: I've just been testing out that approach, running `dbus-launch --autolaunch <machine id>` where `machine-id` is the content of `/var/lib/dbus/machine-id` and this is what I get:

* in a normal terminal (with `DBUS_SESSION_BUS_ADDRESS` set correctly) `dbus-launch --autolaunch` comes back with the wrong address:

```
$ echo $DBUS_SESSION_BUS_ADDRESS
unix:abstract=/tmp/dbus-l6dLTLrbE6

$ dbus-launch --autolaunch $(cat /var/lib/dbus/machine-id)
DBUS_SESSION_BUS_ADDRESS=unix:abstract=/tmp/dbus-2JfLeHwsBS,guid=0baef8fabd37fcc3fc3eee665bcf37e8
DBUS_SESSION_BUS_PID=11748
DBUS_SESSION_BUS_WINDOWID=96468993
```

* run from cron (without `DBUS_SESSION_BUS_ADDRESS` set) I get the error:

```
* * * * * /usr/bin/dbus-launch --autolaunch 5c1047a00114420f9acaf6970395e096 > ~/dbus-launch-autolaunch.txt 2>&1
```

```
$ cat ~/dbus-launch-autolaunch.txt 
Autolaunch error: X11 initialization failed.
```

So it seems this approach isn't going to work - it seems the man page discourages its use, too:

> --autolaunch=MACHINEID
>
> This option implies that dbus-launch should scan for a previously-started session and reuse the values found there. If no session is found, it will start a new session. The --exit-with-session option is implied if --autolaunch is given. This option is for the exclusive use of libdbus, you do not want to use it manually. It may change in the future.